lib/matplotlib/backend_tools.py

```diff
@@ -1017,6 +1017,47 @@ def _mouse_move(self, event):
         self.toolmanager.canvas.draw_idle()
 
 
+class HelpTool(ToolToggleBase):
+    description = 'Print tool list and shortcuts'
+    default_keymap = rcParams['keymap.help']
+    image = 'help.png'
+
+    def __init__(self, *args):
+        ToolToggleBase.__init__(self, *args)
+        self.text_axes = None
+
+    def enable(self, *args):
+        self.text_axes = self.figure.add_axes((0, 0, 1, 1))
+        t = self.get_text()
+        self.text_axes.text(0., 1., t,
+                            va='top',
+                            family='monospace',
+                            backgroundcolor='w',
+                            alpha=1.0)
+        self.figure.canvas.draw_idle()
+
+    def disable(self, *args):
+        self.text_axes.remove()
+        self.figure.canvas.draw_idle()
+
+    def get_text(self):
+        sep = '_' * 80 + '\n'
+        t = sep
+        t += "{0:15} {1:25} {2}\n".format(
```

Comment:
I mean replace 15 and 25 by whatever number is necessary (just count the characters of the longest string).